modules/associate/associate.js

```diff
@@ -1,40 +1,41 @@
-exports.help = function(commandPrefix) {
-	return [[commandPrefix + 'associate "<hook>" "<text>" or clear','Associate and disassociate a phrase with another.']];
-};
-
-exports.match = function(text, commandPrefix, thread, senderName, api) {
-	if (text.startsWith(commandPrefix + 'associate')) {
-		return true;
-	}
-
-	var s = text.toLowerCase();
-	for (var assoc in this.config[thread]) {
-		if (s.indexOf(assoc.toLowerCase()) !== -1) {
-			if (api) {
-				api.sendMessage(this.config[thread][assoc], thread);
-			}
-			return true;
+exports.match = function (event, commandPrefix) {
+    if (event.arguments[0] === commandPrefix + 'associate') {
+        return true;
+    }
+    
+	var s = event.body.toLowerCase();
+	for (var assoc in exports.config[event.thread_id]) {
+        if (s.indexOf(assoc.toLowerCase()) !== -1) {
+            if (!event.__associateCmd) {
+                event.__associateCmd = {
+                    thread: event.thread_id,
+                    responses: []
+                };
+            }
+            event.__associateCmd.responses.push(exports.config[event.thread_id][assoc]);
 		}
 	}
-	return false;
+	return !!event.__associateCmd;
 };
 
-exports.toggleAssociation = function(thread, hook, text) {
-	hook = hook.toLowerCase();
-	if (this.config[thread] && this.config[thread][hook] && !text) {
-		delete this.config[thread][hook];
-		return false;
-	}
+toggleAssociation = function(thread, hook, text) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["toggleAssociation" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529926667/issues/source?bid=3282987&fileBranchId=3282998#l21)